src/server/transaction.cc

```diff
@@ -490,15 +489,22 @@ bool Transaction::RunInShard(EngineShard* shard, bool txq_ooo) {
   }
 
   /*************************************************************************/
+  // at least the coordinator thread owns the reference.
+  DCHECK_GE(GetUseCount(), 1u);
+
+  shard->db_slice().OnCbFinish();
+
+  // Handle result flags to alter behaviour.
+  if (result.flags & RunnableResult::AVOID_CONCLUDING) {
+    CHECK_EQ(unique_shard_cnt_, 1u);  // multi shard must know it ahead, so why do those tricks?
```

Comment:
I do not understand the comment. what tricks? what multi-shard must know?

Reply:
Changed the comment, multi shard must conclude either all or none, because the cbs don't coordinate it must know it ahead -> no point in using this flag